log/log.go

```diff
@@ -243,9 +273,32 @@ func output(w io.Writer, format string, a ...interface{}) {
 // safeWrite is run as a goroutine. It pulls a message from the
 // channel and perform the write.
 func safeWrite() {
-	for ln := range l.write {
-		ln.w.Write(ln.b)
-		atomic.AddInt32(&l.pendingWrites, -1)
+	l.bulkTimer.Reset(GetBulkLogPeriod())
+
+	flush := func() {
+		for k, v := range l.bulkLines {
+			go k.Write(v)
```

Comment:
* I think the use of the goroutine per line could cause lines to be written out-of-order.
* Is it ok to ignore any `Write` errors?
* I think this is causing a datarace when running the tests:
```
==================
WARNING: DATA RACE
Read at 0x00c4200da000 by main goroutine:
  github.com/Comcast/go-log/log_test.ExampleUplevel_Start()
      /usr/local/Cellar/go/1.9.2/libexec/src/bytes/buffer.go:61 +0x26e
  testing.runExample()
      /usr/local/Cellar/go/1.9.2/libexec/src/testing/example.go:122 +0x2a1
  testing.runExamples()
      /usr/local/Cellar/go/1.9.2/libexec/src/testing/example.go:46 +0x1b5
  testing.(*M).Run()
      /usr/local/Cellar/go/1.9.2/libexec/src/testing/testing.go:922 +0x307
  main.main()
      github.com/Comcast/go-log/log/_test/_testmain.go:108 +0x1d3

Previous write at 0x00c4200da000 by goroutine 10:
  bytes.(*Buffer).grow()
      /usr/local/Cellar/go/1.9.2/libexec/src/bytes/buffer.go:135 +0x3c3
  bytes.(*Buffer).Write()
      /usr/local/Cellar/go/1.9.2/libexec/src/bytes/buffer.go:163 +0x16d

Goroutine 10 (finished) created at:
  github.com/Comcast/go-log/log.safeWrite.func1()
      /Users/adam/src/github.com/Comcast/go-log/log/log.go:280 +0x139
  github.com/Comcast/go-log/log.safeWrite()
      /Users/adam/src/github.com/Comcast/go-log/log/log.go:295 +0x4d3
==================
```

Reply:
1. Yes, it's ok to have message out-of-order. Every message has a timestamp so the order can be reconstructed.
2. Is it ok to ignore `Write` errors?  Tricky question.  Would you recommend we check for the error and `Fprintf` to `Stderr`?
3. Yes, I saw datarace, however, few things:
  - It's a tricky situation - I think `go test` (with or without the `-race`) attempts to use the same buffers that we are using for stdout/err display so we get dataraces
  - We had datarace before this change which is due to the same condition I just described.
  - Since we are using a single thread for all writes, the datarace should not happen in a non-test-with-race environment.
  - Can you think of a way to avoid the datarace?